src/frequenz/sdk/microgrid/grid.py

```diff
@@ -0,0 +1,82 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Grid connection point.
+
+This module provides the `Grid` type, which represents a grid connection point
+in a microgrid.
+"""
+
+import logging
+from dataclasses import dataclass
+from typing import Iterable, Optional
+
+from .component import Component
+from .component._component import ComponentCategory
+from .fuse import Fuse
+
+
+@dataclass(frozen=True)
+class Grid:
+    """A grid connection point.
+
+    Attributes:
+        fuse: The fuse protecting the grid connection point.
+    """
+
+    fuse: Fuse
+
+
+_GRID: Optional[Grid] = None
+
+
+def initialize(components: Iterable[Component]) -> None:
+    """Initialize the grid connection.
+
+    Args:
+        components: The components in the microgrid.
+
+    Raises:
+        RuntimeError: If there is more than 1 grid connection point in the
+            microgrid.
+    """
+    global _GRID  # pylint: disable=global-statement
+
+    grid_connections = list(
+        component
+        for component in components
+        if component.category == ComponentCategory.GRID
+    )
+
+    grid_connections_count = len(grid_connections)
+
+    if grid_connections_count == 0:
+        logging.info(
+            "No grid connection found for this microgrid. This is normal for an islanded microgrid."
+        )
+    elif grid_connections_count > 1:
+        raise RuntimeError(
+            f"Expected at most one grid connection, got {len(grid_connections)}"
+        )
+    else:
+        if grid_connections[0].metadata is None:
+            raise RuntimeError("Grid metadata is None")
```

Comment:
Optionally a more "pythonic" way to do it is `if not grid_connections[0].metadata:`

Reply:
Same: https://github.com/frequenz-floss/frequenz-sdk-python/pull/610#discussion_r1307389436

> Oof that's a weird one. It feels like casting fuse to a bool type.